### PR TITLE
Expose JsHasOwnProperty

### PR DESCRIPTION
**Pull request by kfarnung**
Enable support for usage in node's n-api. Added a new unit test for
the functionality.

### CODE REVIEW

**Comment by kfarnung** [comment]
@obastemur This may need to be updated based on the outcome of #3207.  I just followed the existing pattern for `JsHasProperty` before I saw that PR.

File: lib/Jsrt/Jsrt.cpp [hunk 1]
```diff
@@ -4731,4 +4731,21 @@ CHAKRA_API JsGetAndClearExceptionWithMetadata(_Out_ JsValueRef *metadata)
...
+
+CHAKRA_API JsHasOwnProperty(_In_ JsValueRef object, _In_ JsPropertyIdRef propertyId, _Out_ bool *hasOwnProperty)
+{
+    return ContextAPIWrapper<true>([&] (Js::ScriptContext *scriptContext, TTDRecorder& _actionEntryPopper) -> JsErrorCode {
+        PERFORM_JSRT_TTD_RECORD_ACTION(scriptContext, RecordJsRTHasOwnProperty, (Js::PropertyRecord *)propertyId, object);
+
+        VALIDATE_INCOMING_OBJECT(object, scriptContext);
+        VALIDATE_INCOMING_PROPERTYID(propertyId);
+        PARAM_NOT_NULL(hasOwnProperty);
+        *hasOwnProperty = nullptr;
```
Comment (kunalspathak) [inline lib/Jsrt/Jsrt.cpp]: > *hasOwnProperty = nullptr; [](start = 7, length = 27)

shouldn't be initialized to false? I know at other places these variables are initialized to nullptr, but it should be false in those cases as well.

**Comment by obastemur** [comment]
LGTM

File: lib/Jsrt/Jsrt.cpp (hunk 1 above)
Comment (kfarnung) [inline lib/Jsrt/Jsrt.cpp]: Ah, yeah, that came up in  @obastemur PR as well, I'll take a quick scan and fix them up.

File: lib/Runtime/Debug/TTEvents.h [hunk 2]
```diff
@@ -169,7 +169,8 @@ namespace TTD
             CodeParseActionTag,
             CallExistingFunctionActionTag,
 
-            Count
+            Count,
+            HasOwnPropertyActionTag
```
Comment (kfarnung) [inline lib/Runtime/Debug/TTEvents.h]: That was dumb.

**Comment by kfarnung** [comment]
@mrkmarron Can you quickly sanity check my TTD changes?  I'm going to need to write a new test to validate them.

**Comment by mrkmarron** [comment]
From a quick look I would say the changes look good. Let me know if you want me to look at anything specific.

Mark

Get Outlook for iOS<https://aka.ms/o0ukef>
_____________________________

**Comment by liminzhu** [comment]
LGTM

**Comment by kfarnung** [comment]
Added a page to the wiki and updated the reference list:

https://github.com/Microsoft/ChakraCore/wiki/JsHasOwnProperty

/cc @kunalspathak @liminzhu

**Comment by liminzhu** [comment]
@kfarnung awesome! I added an experimental note to the wiki temporarily. Seems like a great addition to Chakra APIs as well and this is just in case Windows team has some feedback along the process.

**Pull request merged**